app/src/pages/PromLens.tsx

```diff
@@ -191,6 +191,18 @@ const PromLens: FC<PathPrefixProps> = ({ pathPrefix }) => {
           );
         }
 
+        // Override Prometheus server from URL for proxy access
+        if (queryParams.ds) {
+          store.dispatch(
+            setServerSettings({
```

Comment:
I think you'll want to use `grafanaDatasourceToServerSettings(Number(queryParams.ds))` here to also populate `url` and `withCredentials` fields for this datasource, like it's done for the default Grafana datasource here: https://github.com/prometheus/promlens/blob/main/app/src/pages/PromLens.tsx#L177

Also, what's the current behavior when an invalid datasource ID is passed in? Would be nice to catch this and make sure that it doesn't just randomly break things but shows an error.

Reply:
Now using grafanaDatasourceToServerSettings. 

The new custom error message is now displayed if a datasource ID is provided that does not exist, or a non number is provided.